multimap/src/main/java/org/infinispan/multimap/impl/function/RemoveFunction.java

```diff
@@ -0,0 +1,93 @@
+package org.infinispan.multimap.impl.function;
+
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.NoSuchElementException;
+import java.util.Set;
+
+import org.infinispan.commons.api.functional.EntryView;
+import org.infinispan.commons.marshall.AdvancedExternalizer;
+import org.infinispan.multimap.impl.ExternalizerIds;
+
+/**
+ * The adding function to remove a ket or a key/value on the multimap
+ *
+ * @author Katia Aresti - karesti@redhat.com
+ * @since 9.1
+ */
+public final class RemoveFunction<K, V> extends BaseFunction<K, V, Boolean> {
+
+   public static final AdvancedExternalizer<RemoveFunction> EXTERNALIZER = new Externalizer();
+   private final K key;
+   private final V value;
+
+   public RemoveFunction(K key) {
+      this.key = key;
+      this.value = null;
+   }
+
+   public RemoveFunction(K key, V value) {
+      this.key = key;
+      this.value = value;
+   }
+
+   @Override
+   public Boolean apply(EntryView.ReadWriteEntryView<K, Collection<V>> entryView) {
+      boolean removed = false;
+      try {
+         // Raises NoSuchElementException if the key/value does not exist
+         Collection<V> values = entryView.get();
+
+         if(value == null) {
+            // Remove the key
+            entryView.remove();
+            removed = true;
+
+         } else if (values.contains(value)){
```

Comment:
This is just to prevent ConcurrentModificationException ? Cause as soon as you call  `entryView.set` wouldn't that replace the current value even without tx?